### PR TITLE
chore(TDI-48645) Add coverage plugin in Jenkins job

### PR DESCRIPTION
**Pull request by acatoire**
chore(TDI-48645) Add coverage plugin in Jenkins job

https://jira.talendforge.org/browse/TDI-48645

Job Done:
- Add the plugin "publishCoverage" using adapters "jacocoAdapter" in build job
- Job name and description has been improved at the same time
- Some code cleaning/grouping
- Add an optional Jenkins debug step
- Add coverage result as artefacts

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -109,31 +112,23 @@ pipeline {
...
-                defaultValue: devNexusRepository,
-                description: 'The Nexus repositories where maven snapshots are deployed.')
+        choice(name: 'Action',
+          choices: ['STANDARD', 'RELEASE', 'DEPLOY'],
+          description: '''
+            Kind of run:
+            STANDARD : (default) classical CI
+            RELEASE : Build release, deploy to the Nexus for master/maintenance branches
+            DEPLOY : Build snapshot, deploy it to the Nexus for any branch''')
+        booleanParam(name: 'SONAR_ANALYSIS', defaultValue: true,
```
Comment (fdesiles) [inline Jenkinsfile:122]: Could we review this part for readability purpose ?

**Review by fdesiles — COMMENTED** [review]
Comment on readability to review.

File: Jenkinsfile (hunk 1 above)
Comment (acatoire) [reply Jenkinsfile:122]: It was to uniform with component runtime, and try to limit the file length,
Those parameters are really rarely edited, and they were taking a lot of place before.
In editor, written like this, you have quickly the name and default value. 
image

**Review by fdesiles — APPROVED** [review]
LGTM

**Pull request merged**